core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1099,26 +1098,17 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
               val result = runLoop(fold.first, stackIndex, stackIndex, currentDepth + 1, ops)
               ops += 1
 
-              if (null eq result)
-                return null
-              else {
-                stackIndex -= 1
-                popStackFrame(stackIndex)
-
-                result match {
-                  case s: Success[Any] =>
-                    cur = fold.successK(s.value)
+              if (null eq result) return null
 
-                  case f: Failure[Any] =>
-                    val cause = f.cause
-                    if (shouldInterrupt()) {
-                      cur = Exit.Failure(cause.stripFailures)
-                    } else {
-                      val f = fold.failureK
+              stackIndex -= 1
+              popStackFrame(stackIndex)
 
-                      cur = f(cause)
-                    }
-                }
+              result match {
+                case s: Success[Any] => cur = fold.successK(s.value)
+                case f: Failure[Any] =>
+                  val cause = f.cause
+                  if (shouldInterrupt()) cur = Exit.Failure(cause.stripFailures)
```

Comment:
Why do we need to strip the failure at all?

Reply:
No idea. That's a good question 🙂